kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -56,6 +56,39 @@ class RobotState;
 namespace kinematics
 {
 
+/*
+ * @enum KinematicSearches
+ * @brief Flags for choosing the type discretization applied on the redundancy during a multiple ik solution search
+ */
+namespace KinematicSearches
+{
+  enum KinematicSearch
+  {
+    ONE = 1,
+    ALL_DISCRETIZED ,
+    SOME_DISCRETIZED,
+    RANDOM_SAMPLE,
+    ALL
```

Comment:
While it's true that most plugins will hide the details of how the redundant joint are manipulated, there was a lot of interest in exposing this capability at the interface in the discussion which led to this PR.  
I could envision a case where in the interest of time one could try to get fewer solutions by discretizing fewer joints while in another case during path planning you'd like to explore larger solution sets in order to generate smoother transitions between points.  These larger sets would of course require more time to generate since more or all redundant joints would be discretized.